api/v1alpha1/akodeploymentconfig_webhook.go

```diff
@@ -75,7 +79,32 @@ func (r *AKODeploymentConfig) ValidateUpdate(old runtime.Object) error {
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type
 func (r *AKODeploymentConfig) ValidateDelete() error {
 	akoDeploymentConfigLog.Info("validate delete", "name", r.Name)
-	return nil
+	// should not delete the akodeploymentconfig selects management cluster
+	if r.Name == ManagementClusterAkoDeploymentConfig {
+		return field.Invalid(field.NewPath("spec", "ClusterSelector"),
+			r.Spec.ClusterSelector,
+			"can't delete akodeploymentconfig object for management cluster")
+	}
+	// should not delete akodeploymentconfig which is used by any clusters
```

Comment:
I feel like enforcing this will cause us troubles.

If we require the ADC selector to be immutable, what if an ADC is selecting clusters by a label `foo: bar`. The only way to delete this ADC is to modify this cluster label.

However, what if `foo: bar` is managed by other controller and we don't have choice to drop it? In this scenarios we can't remove the ADC and disable Avi for this cluster anymore

Reply:
That's a good call, let me remove it.